In [5]:
import sys
sys.executable

'C:\\Users\\DELL\\Downloads\\ItineraireVacances\\.venv\\Scripts\\python.exe'

# Dataset — tripadvisor_european_restaurants
(679.68 MB, 1 file, 42 columns, 5774 downloads, Updated 5 years ago)


https://www.kaggle.com/datasets/stefanoleone992/tripadvisor-european-restaurants

# 1. Charger les données

In [6]:
import pandas as pd
import numpy as np
import pyarrow
from IPython.display import display
import folium
from folium.plugins import MarkerCluster

In [7]:
#lire CSV_PATH (première fois)
CSV_PATH = "tripadvisor_european_restaurants.csv"
df_trip = pd.read_csv(CSV_PATH, sep=',', encoding='utf-8', low_memory = False)

#Convertir en Parquet --> Fichier beaucoup plus léger --> Lecture rapide
PARQUET_PATH = "tripadvisor_european_restaurants.parquet"
df_trip.to_parquet(PARQUET_PATH, compression = "snappy")

#Libère la mémoire et recharger le parquet
del df_trip
df_trip = pd.read_parquet("tripadvisor_european_restaurants.parquet")

#Afficher 2 lignes et TOUTES les colonnes
pd.set_option("display.max_columns", None)
display(df_trip.head(2))

print("\n-------------------------------------")
print("Nombre de lignes et de colones")
print(df_trip.shape)

print("\n-------------------------------------")
print("Display info sur les colonnes et datatype")
display(df_trip.info())

print("\n-------------------------------------")
print("Lister + count les pays de la colonne country")
display(df_trip["country"].value_counts())

print("\n-------------------------------------")
print("Lister + count les régions en France")
display(df_trip.loc[df_trip["country"] == "France", "region"].value_counts())

restaurant_link   restaurant_name  \
0  g10001637-d10002227            Le 147   
1  g10001637-d14975787  Le Saint Jouvent   

                                   original_location country  \
0  ["Europe", "France", "Nouvelle-Aquitaine", "Ha...  France   
1  ["Europe", "France", "Nouvelle-Aquitaine", "Ha...  France   

               region      province           city  \
0  Nouvelle-Aquitaine  Haute-Vienne  Saint-Jouvent   
1  Nouvelle-Aquitaine  Haute-Vienne  Saint-Jouvent   

                                            address   latitude  longitude  \
0       10 Maison Neuve, 87510 Saint-Jouvent France  45.961674   1.169131   
1  16 Place de l Eglise, 87510 Saint-Jouvent France  45.957040   1.205480   

     claimed awards                   popularity_detailed  \
0    Claimed   None  #1 of 2 Restaurants in Saint-Jouvent   
1  Unclaimed   None  #2 of 2 Restaurants in Saint-Jouvent   

                       popularity_generic            top_tags price_level  \
0  #1 of 2 places to eat in Saint-Jouvent  Cheap Eats, French           €   
1  #2 of 2 places to eat in Saint-Jouvent          Cheap Eats           €   

  price_range          meals cuisines special_diets  \
0        None  Lunch, Dinner   French          None   
1        None           None     None          None   

                                            features vegetarian_friendly  \
0  Reservations, Seating, Wheelchair Accessible, ...                   N   
1                                               None                   N   

  vegan_options gluten_free original_open_hours  open_days_per_week  \
0             N           N                None                 NaN   
1             N           N                None                 NaN   

   open_hours_per_week  working_shifts_per_week  avg_rating  \
0                  NaN                      NaN         4.0   
1                  NaN                      NaN         4.0   

   total_reviews_count default_language  reviews_count_in_default_language  \
0                 36.0          English                                2.0   
1                  5.0    All languages                                5.0   

   excellent  very_good  average  poor  terrible  food  service  value  \
0        2.0        0.0      0.0   0.0       0.0   4.0      4.5    4.0   
1        2.0        2.0      1.0   0.0       0.0   NaN      NaN    NaN   

   atmosphere keywords  
0         NaN     None  
1         NaN     None


-------------------------------------
Nombre de lignes et de colones
(1083397, 42)

-------------------------------------
Display info sur les colonnes et datatype
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1083397 entries, 0 to 1083396
Data columns (total 42 columns):
 #   Column                             Non-Null Count    Dtype  
---  ------                             --------------    -----  
 0   restaurant_link                    1083397 non-null  object 
 1   restaurant_name                    1083396 non-null  object 
 2   original_location                  1083397 non-null  object 
 3   country                            1083397 non-null  object 
 4   region                             1033074 non-null  object 
 5   province                           742765 non-null   object 
 6   city                               682712 non-null   object 
 7   address                            1083397 non-null  object 
 8   latitude                           1067607 non-null  floa

None


-------------------------------------
Lister + count les pays de la colonne country


country
Italy               224763
Spain               157479
France              155288
England             144681
Germany             115333
Greece               33763
Portugal             32592
The Netherlands      29792
Poland               24698
Belgium              23711
Austria              20487
Sweden               18555
Czech Republic       14844
Scotland             14215
Ireland              11203
Denmark               9485
Wales                 9134
Croatia               8375
Romania               7842
Hungary               7431
Finland               7372
Bulgaria              4469
Slovakia              4251
Northern Ireland      3634
Name: count, dtype: int64


-------------------------------------
Lister + count les régions en France


region
Ile-de-France                 31271
Auvergne-Rhone-Alpes          20753
Provence-Alpes-Cote d'Azur    19925
Occitanie                     17775
Nouvelle-Aquitaine            15542
Grand Est                      9885
Hauts-de-France                7889
Brittany                       7568
Pays de la Loire               6527
Normandy                       6096
Bourgogne-Franche-Comte        5290
Centre-Val de Loire            4230
Corsica                        2535
Name: count, dtype: int64

# 2. Décrire les données  (région Ile-de-France)

### 2.1. Extraire les données de la région Ile-de-France

In [8]:
#créer un DataFrame filtré avec country = "France" ET region = "Ile-de-France"
df_trip_idf = df_trip[(df_trip["country"] == "France") & (df_trip["region"] == "Ile-de-France")].copy()
print("Nombre de lignes et de colones")
print(df_trip_idf.shape, "\n")
df_trip_idf.head(3)

Nombre de lignes et de colones
(31271, 42) 



restaurant_link      restaurant_name  \
74   g1006520-d1007522            L'imprevu   
75  g1006520-d10128445  La Maison du Poulet   
76   g1006520-d1017687         Little Tokyo   

                                    original_location country         region  \
74  ["Europe", "France", "Ile-de-France", "Yveline...  France  Ile-de-France   
75  ["Europe", "France", "Ile-de-France", "Yveline...  France  Ile-de-France   
76  ["Europe", "France", "Ile-de-France", "Yveline...  France  Ile-de-France   

    province                      city  \
74  Yvelines  Conflans Sainte Honorine   
75  Yvelines  Conflans Sainte Honorine   
76  Yvelines  Conflans Sainte Honorine   

                                              address   latitude  longitude  \
74  9 quai de la Republique, 78700 Conflans Sainte...  48.991110   2.088753   
75  91 avenue Carnot, 78700 Conflans Sainte Honori...  49.000570   2.093569   
76  2 boulevard Troussel, 78700 Conflans Sainte Ho...  48.990112   2.082339   

      claimed                                             awards  \
74    Claimed  Travellers' Choice, Certificate of Excellence ...   
75    Claimed  Travellers' Choice, Certificate of Excellence ...   
76  Unclaimed  Certificate of Excellence 2016, Certificate of...   

                                  popularity_detailed  \
74  #10 of 64 Restaurants in Conflans Sainte Honorine   
75   #3 of 64 Restaurants in Conflans Sainte Honorine   
76  #13 of 64 Restaurants in Conflans Sainte Honorine   

                                   popularity_generic  \
74  #10 of 68 places to eat in Conflans Sainte Hon...   
75  #3 of 68 places to eat in Conflans Sainte Hono...   
76  #13 of 68 places to eat in Conflans Sainte Hon...   

                                  top_tags price_level price_range  \
74                     Fine Dining, French        €€€€        None   
75  Mid-range, French, Vegetarian Friendly      €€-€€€        None   
76       Mid-range, Japanese, Sushi, Asian      €€-€€€        None   

                         meals                cuisines        special_diets  \
74  Lunch, Dinner, After-hours                  French                 None   
75               Lunch, Dinner                  French  Vegetarian Friendly   
76               Lunch, Dinner  Japanese, Sushi, Asian  Vegetarian Friendly   

                                             features vegetarian_friendly  \
74  Reservations, Seating, Serves Alcohol, Accepts...                   N   
75                                               None                   Y   
76                                               None                   Y   

   vegan_options gluten_free  \
74             N           N   
75             N           N   
76             N           N   

                                  original_open_hours  open_days_per_week  \
74  {"Mon": ["10:00-15:00", "18:00-23:00"], "Tue":...                 7.0   
75  {"Mon": [], "Tue": [], "Wed": ["12:00-14:00", ...                 5.0   
76  {"Mon": ["11:30-02:30"], "Tue": ["11:30-02:30"...                 7.0   

    open_hours_per_week  working_shifts_per_week  avg_rating  \
74                 60.0                     13.0         4.0   
75                 20.0                      9.0         4.5   
76                105.0                      7.0         4.0   

    total_reviews_count default_language  reviews_count_in_default_language  \
74                292.0          English                                2.0   
75                296.0    All languages                              296.0   
76                204.0          English                                6.0   

    excellent  very_good  average  poor  terrible  food  service  value  \
74        1.0        1.0      0.0   0.0       0.0   4.0      4.0    3.5   
75      202.0       73.0     18.0   1.0       2.0   4.5      4.5    4.5   
76        3.0        2.0      1.0   0.0       0.0   4.0      4.0    3.5   

    atmosphere keywords  
74         3.0     None  
75      

### 2.2. Les colonnes Dtype float64

In [9]:
print("\n-------------------------------------")
print("Describe la colonne open_days_per_week (disponibilité globale):")
print("---------------------------------------")
print("ex : 7 = ouvert tous les jours")
display(df_trip_idf["open_days_per_week"].unique())
display(df_trip_idf["open_days_per_week"].describe())

print("\n-------------------------------------")
print("Describe la colonne working_shifts_per_week (flexibilité horaire) :")
print("ex : 6 jours ouverts * 2 services/jours = 12 working shifts")
print("---------------------------------------")
display(df_trip_idf["working_shifts_per_week"].unique())
display(df_trip_idf["working_shifts_per_week"].describe())

print("\n-------------------------------------")
print("Describe la colonne avg_rating (perception globale sur TripAdvisor (1 à 5)) :")
print("---------------------------------------")
display(df_trip_idf["avg_rating"].unique())
display(df_trip_idf["avg_rating"].describe())

print("\n-------------------------------------")
print("Describe la colonne total_reviews_count (nombre total d’avis) (crédibilité de la note ---> notoriété / fréquentation) :")
print("---------------------------------------")
display(df_trip_idf["total_reviews_count"].unique())
display(df_trip_idf["total_reviews_count"].describe())

print("\n-------------------------------------")
print("Describe la colonne reviews_count_in_default_language (nombre d’avis dans la langue principale) :")
print("---------------------------------------")
display(df_trip_idf["reviews_count_in_default_language"].unique()[:10])
display(df_trip_idf["reviews_count_in_default_language"].describe())

print("\n-------------------------------------")
print("Describe la colonne excellent (nombre d’avis par catégorie excellent) :")
print("---------------------------------------")
display(df_trip_idf["excellent"].unique()[:10])
display(df_trip_idf["excellent"].describe())

print("\n-------------------------------------")
print("Describe la colonne very_good (nombre d’avis par catégorie very_good) :")
print("---------------------------------------")
display(df_trip_idf["very_good"].unique()[:10])
display(df_trip_idf["very_good"].describe())

print("\n-------------------------------------")
print("Describe la colonne average (nombre d’avis par catégorie average) :")
print("---------------------------------------")
display(df_trip_idf["average"].unique()[:10])
display(df_trip_idf["average"].describe())

print("\n-------------------------------------")
print("Describe la colonne poor (nombre d’avis par catégorie poor):")
print("---------------------------------------")
display(df_trip_idf["poor"].unique()[:10])
display(df_trip_idf["poor"].describe())

print("\n-------------------------------------")
print("Describe la colonne terrible (nombre d’avis par catégorie terrible):")
print("---------------------------------------")
display(df_trip_idf["terrible"].unique()[:10])
display(df_trip_idf["terrible"].describe())

print("\n-------------------------------------")
print("Describe la colonne food (notes spécifiques (1 à 5) sur food) :")
print("---------------------------------------")
display(df_trip_idf["food"].unique())
display(df_trip_idf["food"].describe())

print("\n-------------------------------------")
print("Describe la colonne service (notes spécifiques (1 à 5) sur service) :")
print("---------------------------------------")
display(df_trip_idf["service"].unique())
display(df_trip_idf["service"].describe())

print("\n-------------------------------------")
print("Describe la colonne value (notes spécifiques (1 à 5) sur value) :")
print("---------------------------------------")
display(df_trip_idf["value"].unique())
display(df_trip_idf["value"].describe())

print("\n-------------------------------------")
print("Describe la colonne atmosphere (notes spécifiques (1 à 5) sur atmosphere) :")
print("---------------------------------------")
display(df_trip_idf["atmosphere"].unique())
display(df_trip_idf["atmosphere"].describe())

#Vérifier les valeurs manquantes des colonnes Dtype float64
print("\n-------------------------------------")
print("Les valeurs manquantes des colonnes Dtype float64 (sur 31271 lignes) :")
print("---------------------------------------")
missing_float64_summary = pd.DataFrame({"missing_count": df_trip_idf.select_dtypes(include="float64").isna().sum(),
                                        "missing_pct": (df_trip_idf.select_dtypes(include="float64").isna().sum() / 31271 * 100).round(2)}
                                      ).sort_values("missing_pct", ascending=False)
display(missing_float64_summary)


-------------------------------------
Describe la colonne open_days_per_week (disponibilité globale):
---------------------------------------
ex : 7 = ouvert tous les jours


array([ 7.,  5., nan,  6.,  1.,  4.,  2.,  3.])

count    17983.000000
mean         6.250014
std          0.888699
min          1.000000
25%          6.000000
50%          6.000000
75%          7.000000
max          7.000000
Name: open_days_per_week, dtype: float64


-------------------------------------
Describe la colonne working_shifts_per_week (flexibilité horaire) :
ex : 6 jours ouverts * 2 services/jours = 12 working shifts
---------------------------------------


array([13.,  9.,  7., 12., nan, 10.,  2., 14.,  8.,  6., 11.,  4.,  5.,
        1.,  3., 15.])

count    17983.000000
mean         8.642941
std          3.034872
min          1.000000
25%          6.000000
50%          7.000000
75%         12.000000
max         15.000000
Name: working_shifts_per_week, dtype: float64


-------------------------------------
Describe la colonne avg_rating (perception globale sur TripAdvisor (1 à 5)) :
---------------------------------------


array([4. , 4.5, 3.5, 3. , nan, 5. , 2.5, 2. , 1. , 1.5])

count    28237.000000
mean         3.930269
std          0.757283
min          1.000000
25%          3.500000
50%          4.000000
75%          4.500000
max          5.000000
Name: avg_rating, dtype: float64


-------------------------------------
Describe la colonne total_reviews_count (nombre total d’avis) (crédibilité de la note ---> notoriété / fréquentation) :
---------------------------------------


array([292., 296., 204., ..., 759., 832., 822.], shape=(1132,))

count    29209.000000
mean        99.762299
std        261.139667
min          0.000000
25%          6.000000
50%         27.000000
75%         98.000000
max      15188.000000
Name: total_reviews_count, dtype: float64


-------------------------------------
Describe la colonne reviews_count_in_default_language (nombre d’avis dans la langue principale) :
---------------------------------------


array([  2., 296.,   6.,  17.,   1.,  57.,  33., 224.,  15.,   9.])

count    28258.000000
mean        27.505804
std        101.198160
min          1.000000
25%          2.000000
50%          6.000000
75%         19.000000
max       5090.000000
Name: reviews_count_in_default_language, dtype: float64


-------------------------------------
Describe la colonne excellent (nombre d’avis par catégorie excellent) :
---------------------------------------


array([  1., 202.,   3.,   5.,   0.,   6.,   9., 155.,  10.,   8.])

count    28258.000000
mean        13.220752
std         51.438351
min          0.000000
25%          1.000000
50%          3.000000
75%          9.000000
max       2662.000000
Name: excellent, dtype: float64


-------------------------------------
Describe la colonne very_good (nombre d’avis par catégorie very_good) :
---------------------------------------


array([ 1., 73.,  2.,  5., 18.,  0.,  8., 60.,  4., nan])

count    28258.000000
mean         7.869524
std         29.286814
min          0.000000
25%          0.000000
50%          2.000000
75%          5.000000
max       1420.000000
Name: very_good, dtype: float64


-------------------------------------
Describe la colonne average (nombre d’avis par catégorie average) :
---------------------------------------


array([ 0., 18.,  1.,  2., 11.,  7.,  4., nan,  6.,  3.])

count    28258.000000
mean         2.890261
std         13.239137
min          0.000000
25%          0.000000
50%          0.000000
75%          2.000000
max        706.000000
Name: average, dtype: float64


-------------------------------------
Describe la colonne poor (nombre d’avis par catégorie poor):
---------------------------------------


array([ 0.,  1., 10.,  6.,  2.,  9., nan,  4., 12.,  3.])

count    28258.000000
mean         1.510121
std          6.953421
min          0.000000
25%          0.000000
50%          0.000000
75%          1.000000
max        422.000000
Name: poor, dtype: float64


-------------------------------------
Describe la colonne terrible (nombre d’avis par catégorie terrible):
---------------------------------------


array([ 0.,  2.,  4., 12.,  3.,  1., nan,  6., 17.,  8.])

count    28258.000000
mean         2.015146
std          8.916182
min          0.000000
25%          0.000000
50%          0.000000
75%          1.000000
max        443.000000
Name: terrible, dtype: float64


-------------------------------------
Describe la colonne food (notes spécifiques (1 à 5) sur food) :
---------------------------------------


array([4. , 4.5, nan, 3. , 3.5, 5. , 2.5, 2. , 1.5, 1. ])

count    18407.000000
mean         3.983973
std          0.605862
min          1.000000
25%          3.500000
50%          4.000000
75%          4.500000
max          5.000000
Name: food, dtype: float64


-------------------------------------
Describe la colonne service (notes spécifiques (1 à 5) sur service) :
---------------------------------------


array([4. , 4.5, 3. , nan, 3.5, 5. , 2.5, 2. , 1.5, 1. ])

count    18491.000000
mean         3.982370
std          0.626689
min          1.000000
25%          3.500000
50%          4.000000
75%          4.500000
max          5.000000
Name: service, dtype: float64


-------------------------------------
Describe la colonne value (notes spécifiques (1 à 5) sur value) :
---------------------------------------


array([3.5, 4.5, 4. , nan, 3. , 5. , 2.5, 2. , 1.5, 1. ])

count    18453.000000
mean         3.845472
std          0.611979
min          1.000000
25%          3.500000
50%          4.000000
75%          4.500000
max          5.000000
Name: value, dtype: float64


-------------------------------------
Describe la colonne atmosphere (notes spécifiques (1 à 5) sur atmosphere) :
---------------------------------------


array([3. , nan, 3.5, 2.5, 4.5, 4. , 2. , 1.5, 5. , 1. ])

count    8246.000000
mean        3.766008
std         0.535208
min         1.000000
25%         3.500000
50%         4.000000
75%         4.000000
max         5.000000
Name: atmosphere, dtype: float64


-------------------------------------
Les valeurs manquantes des colonnes Dtype float64 (sur 31271 lignes) :
---------------------------------------


missing_count  missing_pct
atmosphere                                 23025        73.63
working_shifts_per_week                    13288        42.49
open_days_per_week                         13288        42.49
open_hours_per_week                        13288        42.49
food                                       12864        41.14
value                                      12818        40.99
service                                    12780        40.87
avg_rating                                  3034         9.70
terrible                                    3013         9.64
average                                     3013         9.64
poor                                        3013         9.64
reviews_count_in_default_language           3013         9.64
excellent                                   3013         9.64
very_good                                   3013         9.64
total_reviews_count                         2062         6.59
latitude                                     173         0.55
longitude                                    173         0.55

### 2.3. Les colonnes Dtype Object

In [10]:
print("\n-------------------------------------")
print("Les valeurs uniques de la colonne claimed (Restaurant revendiqué par le propriétaire) :")
print("---------------------------------------")
display(df_trip_idf["claimed"].unique())

print("\n-------------------------------------")
print("Les valeurs uniques de la colonne awards (Récompenses TripAdvisor (Certificate of Excellence, Travellers’ Choice…)) :")
print("---------------------------------------")
display(df_trip_idf["awards"].unique()[:10])

print("\n-------------------------------------")
print("Les valeurs uniques de la colonne popularity_detailed (Classement local précis) :")
print("---------------------------------------")
display(df_trip_idf["popularity_detailed"].unique())

print("\n-------------------------------------")
print("Les valeurs uniques de la colonne popularity_generic (Classement global “places to eat”) :")
print("---------------------------------------")
display(df_trip_idf["popularity_generic"].unique())

print("\n-------------------------------------")
print("Les valeurs uniques de la colonne top_tags (Combinaison de gamme + cuisines) :")
print("---------------------------------------")
display(df_trip_idf["top_tags"].unique())

print("\n-------------------------------------")
print("Les valeurs uniques de la colonne price_level (Indication simple du prix) :")
print("---------------------------------------")
display(df_trip_idf["price_level"].unique())

print("\n-------------------------------------")
print("Les valeurs uniques de la colonne price_range (Fourchette détaillée) :")
print("---------------------------------------")
display(df_trip_idf["price_range"].unique())

print("\n-------------------------------------")
print("Les valeurs uniques de la colonne meals (Types de repas (Breakfast, Lunch, Dinner…)) :")
print("---------------------------------------")
display(df_trip_idf["meals"].unique()[:10])

print("\n-------------------------------------")
print("Les valeurs uniques de la colonne cuisines (Types de cuisines (liste)) :")
print("---------------------------------------")
display(df_trip_idf["cuisines"].unique())

print("\n-------------------------------------")
print("Les valeurs uniques de la colonne special_diets (Régimes spéciaux (Vegan, Halal…)) :")
print("---------------------------------------")
display(df_trip_idf["special_diets"].unique()[:20])

print("\n-------------------------------------")
print("Les valeurs uniques de la colonne features (Services proposés (wifi, terrasse, réservation…)) :")
print("---------------------------------------")
display(df_trip_idf["features"].unique())

print("\n-------------------------------------")
print("Les valeurs uniques de la colonne vegetarian_friendly (Indicateurs binaires (Y / N) sur label vegetarian_friendly) :")
print("---------------------------------------")
display(df_trip_idf["vegetarian_friendly"].unique())

print("\n-------------------------------------")
print("Les valeurs uniques de la colonne vegan_options (Indicateurs binaires (Y / N) sur label vegan_options) :")
print("---------------------------------------")
display(df_trip_idf["vegan_options"].unique())

print("\n-------------------------------------")
print("Les valeurs uniques de la colonne gluten_free (Indicateurs binaires (Y / N) sur label gluten_free) :")
print("---------------------------------------")
display(df_trip_idf["gluten_free"].unique())

print("\n-------------------------------------")
print("Les valeurs uniques de la colonne original_open_hours (Horaires détaillés (JSON encodé en texte)) :")
print("---------------------------------------")
display(df_trip_idf["original_open_hours"].unique()[:5])

print("\n-------------------------------------")
print("Les valeurs uniques de la colonne default_language :")
print("---------------------------------------")
display(df_trip_idf["default_language"].unique())

print("\n-------------------------------------")
print("Les valeurs uniques de la colonne keywords (Mots-clés extraits des avis) :")
print("---------------------------------------")
display(df_trip_idf["keywords"].unique())

#Vérifier les valeurs manquantes des colonnes Dtype OBJECT
print("\n-------------------------------------")
print("Les valeurs manquantes des colonnes Dtype OBJECT (sur 31271 lignes) :")
print("---------------------------------------")
missing_object_summary = pd.DataFrame({"missing_count": df_trip_idf.select_dtypes(include="object").isna().sum(),
                                        "missing_pct": (df_trip_idf.select_dtypes(include="object").isna().sum() / 31271 * 100).round(2)}
                                      ).sort_values("missing_pct", ascending=False)
display(missing_object_summary)


-------------------------------------
Les valeurs uniques de la colonne claimed (Restaurant revendiqué par le propriétaire) :
---------------------------------------


array(['Claimed', 'Unclaimed', None], dtype=object)


-------------------------------------
Les valeurs uniques de la colonne awards (Récompenses TripAdvisor (Certificate of Excellence, Travellers’ Choice…)) :
---------------------------------------


array(["Travellers' Choice, Certificate of Excellence 2020, Certificate of Excellence 2019",
       "Travellers' Choice, Certificate of Excellence 2020, Certificate of Excellence 2019, Certificate of Excellence 2018, Certificate of Excellence 2017",
       'Certificate of Excellence 2016, Certificate of Excellence 2015',
       None,
       'Certificate of Excellence 2018, Certificate of Excellence 2016',
       'Certificate of Excellence 2015',
       "Travellers' Choice, Certificate of Excellence 2020, Certificate of Excellence 2019, Certificate of Excellence 2018, Certificate of Excellence 2017, Certificate of Excellence 2016, Certificate of Excellence 2015",
       'Certificate of Excellence 2017, Certificate of Excellence 2016',
       'Certificate of Excellence 2019, Certificate of Excellence 2018, Certificate of Excellence 2017, Certificate of Excellence 2016, Certificate of Excellence 2015',
       'Certificate of Excellence 2017, Certificate of Excellence 2016, Certificate of 


-------------------------------------
Les valeurs uniques de la colonne popularity_detailed (Classement local précis) :
---------------------------------------


array(['#10 of 64 Restaurants in Conflans Sainte Honorine',
       '#3 of 64 Restaurants in Conflans Sainte Honorine',
       '#13 of 64 Restaurants in Conflans Sainte Honorine', ...,
       '#1 of 1 Restaurant in Auteuil-le-Roi',
       '#1 of 1 Restaurant in Arnouville-les-Mantes',
       '#1 of 1 Restaurant in Thoury-Ferrottes'],
      shape=(27340,), dtype=object)


-------------------------------------
Les valeurs uniques de la colonne popularity_generic (Classement global “places to eat”) :
---------------------------------------


array(['#10 of 68 places to eat in Conflans Sainte Honorine',
       '#3 of 68 places to eat in Conflans Sainte Honorine',
       '#13 of 68 places to eat in Conflans Sainte Honorine', ...,
       '#1 of 1 places to eat in Auteuil-le-Roi',
       '#1 of 1 places to eat in Arnouville-les-Mantes',
       '#1 of 1 places to eat in Thoury-Ferrottes'],
      shape=(26769,), dtype=object)


-------------------------------------
Les valeurs uniques de la colonne top_tags (Combinaison de gamme + cuisines) :
---------------------------------------


array(['Fine Dining, French', 'Mid-range, French, Vegetarian Friendly',
       'Mid-range, Japanese, Sushi, Asian', ...,
       'American, Steakhouse, Fast food, European',
       'Fine Dining, Italian, Lebanese, Mediterranean',
       'Cheap Eats, French, American, Deli'], shape=(3565,), dtype=object)


-------------------------------------
Les valeurs uniques de la colonne price_level (Indication simple du prix) :
---------------------------------------


array(['€€€€', '€€-€€€', '€', None], dtype=object)


-------------------------------------
Les valeurs uniques de la colonne price_range (Fourchette détaillée) :
---------------------------------------


array([None, '€10-€15', '€15-€35', ..., '€1-€4', '€8-€68', '€30-€37'],
      shape=(1412,), dtype=object)


-------------------------------------
Les valeurs uniques de la colonne meals (Types de repas (Breakfast, Lunch, Dinner…)) :
---------------------------------------


array(['Lunch, Dinner, After-hours', 'Lunch, Dinner', 'Dinner', None,
       'Dinner, Lunch', 'Dinner, Lunch, Drinks', 'Lunch, Dinner, Drinks',
       'Breakfast, Lunch, Dinner', 'Dinner, After-hours, Drinks',
       'Breakfast, Lunch, Dinner, Brunch, After-hours'], dtype=object)


-------------------------------------
Les valeurs uniques de la colonne cuisines (Types de cuisines (liste)) :
---------------------------------------


array(['French', 'Japanese, Sushi, Asian', 'Pakistani, Middle Eastern',
       ...,
       'Lebanese, Mediterranean, Sicilian, Southern-Italian, Italian',
       'Japanese, Sushi, Asian, Chinese, Korean',
       'French, American, Street Food, Deli'], shape=(3841,), dtype=object)


-------------------------------------
Les valeurs uniques de la colonne special_diets (Régimes spéciaux (Vegan, Halal…)) :
---------------------------------------


array([None, 'Vegetarian Friendly', 'Vegetarian Friendly, Vegan Options',
       'Vegetarian Friendly, Halal',
       'Vegetarian Friendly, Vegan Options, Gluten Free Options', 'Halal',
       'Gluten Free Options',
       'Vegan Options, Vegetarian Friendly, Gluten Free Options',
       'Vegan Options', 'Vegan Options, Vegetarian Friendly',
       'Vegetarian Friendly, Gluten Free Options', 'Kosher',
       'Vegetarian Friendly, Vegan Options, Kosher',
       'Vegetarian Friendly, Vegan Options, Halal',
       'Vegetarian Friendly, Kosher',
       'Gluten Free Options, Vegetarian Friendly, Vegan Options',
       'Vegetarian Friendly, Vegan Options, Halal, Gluten Free Options',
       'Vegan Options, Gluten Free Options, Vegetarian Friendly',
       'Vegetarian Friendly, Gluten Free Options, Vegan Options',
       'Vegan Options, Gluten Free Options'], dtype=object)


-------------------------------------
Les valeurs uniques de la colonne features (Services proposés (wifi, terrasse, réservation…)) :
---------------------------------------


array(['Reservations, Seating, Serves Alcohol, Accepts Credit Cards, Table Service, Highchairs Available',
       None,
       'Takeout, Reservations, Seating, Wheelchair Accessible, Table Service',
       ..., 'Takeout, Outdoor Seating, Seating, Reservations',
       'Delivery, Takeout, Seating, Serves Alcohol, Accepts Credit Cards, Reservations, Table Service',
       'Reservations, Seating, Parking Available, Free off-street parking, Serves Alcohol, Accepts Mastercard, Accepts Visa, Free Wifi, Accepts Credit Cards, Table Service, Outdoor Seating, Highchairs Available, Wheelchair Accessible'],
      shape=(2298,), dtype=object)


-------------------------------------
Les valeurs uniques de la colonne vegetarian_friendly (Indicateurs binaires (Y / N) sur label vegetarian_friendly) :
---------------------------------------


array(['N', 'Y'], dtype=object)


-------------------------------------
Les valeurs uniques de la colonne vegan_options (Indicateurs binaires (Y / N) sur label vegan_options) :
---------------------------------------


array(['N', 'Y'], dtype=object)


-------------------------------------
Les valeurs uniques de la colonne gluten_free (Indicateurs binaires (Y / N) sur label gluten_free) :
---------------------------------------


array(['N', 'Y'], dtype=object)


-------------------------------------
Les valeurs uniques de la colonne original_open_hours (Horaires détaillés (JSON encodé en texte)) :
---------------------------------------


array(['{"Mon": ["10:00-15:00", "18:00-23:00"], "Tue": ["10:00-15:00", "19:00-23:00"], "Wed": ["10:00-15:00", "19:00-23:00"], "Thu": ["10:00-15:00", "19:00-23:00"], "Fri": ["10:00-15:00", "19:00-23:00"], "Sat": ["10:00-15:00", "19:00-23:00"], "Sun": ["10:00-15:00"]}',
       '{"Mon": [], "Tue": [], "Wed": ["12:00-14:00", "19:00-21:30"], "Thu": ["12:00-14:00", "19:00-21:30"], "Fri": ["12:00-14:00", "19:00-21:30"], "Sat": ["12:00-14:00", "19:00-21:30"], "Sun": ["12:00-14:00"]}',
       '{"Mon": ["11:30-02:30"], "Tue": ["11:30-02:30"], "Wed": ["11:30-02:30"], "Thu": ["11:30-02:30"], "Fri": ["11:30-02:30"], "Sat": ["11:30-02:30"], "Sun": ["11:30-02:30"]}',
       '{"Mon": ["18:30-22:30"], "Tue": ["11:30-14:30", "18:30-22:30"], "Wed": ["11:30-14:30", "18:30-22:30"], "Thu": ["11:30-14:30", "18:30-22:30"], "Fri": ["11:30-14:30", "18:30-22:30"], "Sat": ["11:30-14:30", "18:30-22:30"], "Sun": ["18:30-22:30"]}',
       None], dtype=object)


-------------------------------------
Les valeurs uniques de la colonne default_language :
---------------------------------------


array(['English', 'All languages', None], dtype=object)


-------------------------------------
Les valeurs uniques de la colonne keywords (Mots-clés extraits des avis) :
---------------------------------------


array([None, 'risotto, salad, antipasti, disney village, fresh food',
       'burger, potatoes, disney village, free refills, large portion of chips',
       ..., 'crepes, ice cream, delicious food, moulin, hotel',
       'indian food, quantity and quality, owner, home, paris',
       'steak, beer, paris, sign, brasserie'], shape=(1221,), dtype=object)


-------------------------------------
Les valeurs manquantes des colonnes Dtype OBJECT (sur 31271 lignes) :
---------------------------------------


missing_count  missing_pct
keywords                     30050        96.10
special_diets                24797        79.30
awards                       24053        76.92
price_range                  20948        66.99
features                     20089        64.24
province                     18138        58.00
original_open_hours          13288        42.49
meals                        12052        38.54
price_level                   7372        23.57
cuisines                      5915        18.92
top_tags                      4007        12.81
popularity_generic            3050         9.75
default_language              3013         9.64
popularity_detailed           2994         9.57
claimed                         35         0.11
city                             4         0.01
restaurant_link                  0         0.00
address                          0         0.00
original_location                0         0.00
country                          0         0.00
region                           0         0.00
restaurant_name                  0         0.00
vegetarian_friendly              0         0.00
vegan_options                    0         0.00
gluten_free                      0         0.00

# 3. Les colonnes à prendre ou à laisser

### Les colonnes à garder abslument  :

**Dtype float**
- Itinéraire carte : latitude (0,5% missing), longitude (0,5% missing)

- Qualité globale : avg_rating (9,70% missing)

- Crédibilité: total_reviews_count (6,59% missing)


**Dtype object** :
- Filtres localisations géographiques : country (0% missing), region (0% missing), city (0,01% missing), address (0% missing)
  
- Identité : restaurant_name (0% missing), restaurant_link (0% missing)

- Statut + informations générales : claimed (0.11% missing), price_level (23.57% missing), price_range (66.99% missing), meals (38.54% missing), top_tags (12.81% missing), original_open_hours (42,49% missing)

- Labels alimentaires simples : vegetarian_friendly (0% missing), vegan_options (0% missing), gluten_free (0% missing)


### Les colonnes à transformer  :

- A transformer en score binaire (0/1) : awards (76.92% missing) 

- A transformer en proportions (%) : excellent (9.64% missing), very_good (9.64% missing), average (9.64%missing)

- A transformer en score moyen partiel : food (41.14% missing), service (40.87% missing), value (40.99% missing)

- A transformer en score d’accessibilité/Disponibilité/Flexibilité: open_days_per_week (42.49% missing)+ working_shifts_per_week (42.49% missing)

- A extraire l'option Halal de la colonne special_diets (79.30% missing) et les transformer en booléan

- A transformer en booléan : "vegetarian_friendly", "vegan_options", "gluten_free"


### Les colonnes à ignorer :
- Beaucoup de valeurs manquantes Dtype Float : atmosphere (74 % missing), open_hours_per_week (42,49% missing)

- Beaucoup de valeurs manquantes Dtype Object : keywords (96%), province (58% missing)

- Bruit: cuisines (18.92% missing), popularity_detailed (9.57% missing), popularity_generic (9.75% missing), original_location (0% missing), reviews_count_in_default_language (9.64% missing), poor (9.64% missing), terrible (9.64% missing), features (64.24% missing), default_language (9.64% missing)

# 4. Nettoyer + compléter + transformer les valeurs

### 4.1. Nettoyer

In [11]:
df_trip_idf_clean = df_trip_idf.copy()

#Supprimer toutes les lignes où latitude OU longitude est manquante (NaN)
df_trip_idf_clean = df_trip_idf_clean.dropna(subset=["latitude", "longitude"]).copy()

### 4.2. Compléter

In [12]:
#Créer une variable indicatrice pour avg_rating dont valeurs manquantes et la compléter avec une valeur neutre pondérée (median) dans une colonne à part pour l'étape score
df_trip_idf_clean["avg_rating_missing"] = (df_trip_idf_clean["avg_rating"].isna()).astype(int)
median_rating = df_trip_idf_clean["avg_rating"].median()
df_trip_idf_clean["avg_rating_filled"] = (df_trip_idf_clean["avg_rating"].fillna(median_rating))

#Créer une variable indicatrice pour total_reviews_count dont valeurs manquantes et la compléter = 0
df_trip_idf_clean["total_reviews_count_missing"] = (df_trip_idf_clean["total_reviews_count"].isna()).astype(int)
df_trip_idf_clean["total_reviews_count"] = df_trip_idf_clean["total_reviews_count"].fillna(0)

#Créer une catégorie explicite Unknown pour les price_level dont valeurs manquantes
df_trip_idf_clean["price_level"] = df_trip_idf_clean["price_level"].fillna("Unknown")

#Hypothèse raisonnable : un restaurant non revendiqué n’a probablement pas fait la démarche
df_trip_idf_clean["claimed"] = df_trip_idf_clean["claimed"].fillna("Unclaimed")

### 4.3. Transformer

#### 4.3.a Colonne "awards" --> score binaire (0/1)

In [13]:
#transformer en score binaire (0/1) : awards (76.92% missing)
df_trip_idf_clean["awards_binary"] = (df_trip_idf_clean["awards"].notna()).astype(int)
df_trip_idf_clean[["awards", "awards_binary"]].head(5)

awards  awards_binary
74  Travellers' Choice, Certificate of Excellence ...              1
75  Travellers' Choice, Certificate of Excellence ...              1
76  Certificate of Excellence 2016, Certificate of...              1
77                                               None              0
78                                               None              0

#### 4.3.b Colonnes "excellent", "very_good", "average" --> proportions (%)

In [14]:
#transformer en proportions (%) : excellent (9.64% missing), very_good (9.64% missing), average (9.64%missing)

#compléter les valeurs manquantes des colonnes "excellent", "very_good", "average" = 0
df_trip_idf_clean[["excellent", "very_good", "average"]] = df_trip_idf_clean[["excellent", "very_good", "average"]].fillna(0)

#les transformer en proportions (%) en basant sur la colonne "total_reviews_count"
df_trip_idf_clean["excellent_pct"] = ((df_trip_idf_clean["excellent"] / df_trip_idf_clean["total_reviews_count"])*100).round(2)
df_trip_idf_clean["very_good_pct"] = ((df_trip_idf_clean["very_good"] / df_trip_idf_clean["total_reviews_count"])*100).round(2)
df_trip_idf_clean["average_pct"] = ((df_trip_idf_clean["average"] / df_trip_idf_clean["total_reviews_count"])*100).round(2)

#vérifier
df_trip_idf_clean[["total_reviews_count","excellent","excellent_pct", "very_good", "very_good_pct", "average","average_pct"]].head(5)

total_reviews_count  excellent  excellent_pct  very_good  very_good_pct  \
74                292.0        1.0           0.34        1.0           0.34   
75                296.0      202.0          68.24       73.0          24.66   
76                204.0        3.0           1.47        2.0           0.98   
77                 17.0        5.0          29.41        5.0          29.41   
78                 12.0        0.0           0.00        1.0           8.33   

    average  average_pct  
74      0.0         0.00  
75     18.0         6.08  
76      1.0         0.49  
77      2.0        11.76  
78      0.0         0.00

#### 4.3.c Colonnes "food", "service", "value" --> score moyen partiel

In [15]:
#transformer en score moyen partiel : food (41.14% missing), service (40.87% missing), value (40.99% missing)

#Créer le score moyen partiel en ignorant automatiquement les NaN + caculant la moyenne sur 1, 2 ou 3 colonnes + retournant NaN seulement si les 3 sont manquantes
df_trip_idf_clean["partial_quality_score"] = ((df_trip_idf_clean[["food", "service", "value"]].mean(axis=1, skipna=True))).round(2)

#Ajouter un indicateur de fiabilité (3=très fiable grace aux valeurs présentes sur les 3 colonnes, 2=fiable, 1=faible, 0=aucune info)
df_trip_idf_clean["partial_quality_count"] = (df_trip_idf_clean[["food", "service", "value"]].notna().sum(axis=1))

#Flager score partiel incomplet (inférieur à 3)
df_trip_idf_clean["partial_quality_missing"] = (df_trip_idf_clean["partial_quality_count"] < 3).astype(int)

#Score pondéré par la fiabilité
df_trip_idf_clean["partial_quality_score_weighted"] = (df_trip_idf_clean["partial_quality_score"] * (df_trip_idf_clean["partial_quality_count"] / 3)).round(2)

#vérifier
df_trip_idf_clean[["food", "service", "value", "partial_quality_score", "partial_quality_count", "partial_quality_score_weighted"]].head(5)

food  service  value  partial_quality_score  partial_quality_count  \
74   4.0      4.0    3.5                   3.83                      3   
75   4.5      4.5    4.5                   4.50                      3   
76   4.0      4.0    3.5                   3.83                      3   
77   4.0      3.0    4.0                   3.67                      3   
78   NaN      NaN    NaN                    NaN                      0   

    partial_quality_score_weighted  
74                            3.83  
75                            4.50  
76                            3.83  
77                            3.67  
78                             NaN

#### 4.3.d Colonnes "open_days_per_week", "working_shifts_per_week" --> score d’accessibilité

In [16]:
#transformer en score d’accessibilité/Disponibilité/Flexibilité: open_days_per_week (42.49% missing)+ working_shifts_per_week (42.49% missing)

#Créer un indicateurs sur les valeurs manquantes
df_trip_idf_clean["open_days_missing"] = (df_trip_idf_clean["open_days_per_week"].isna().astype(int))
df_trip_idf_clean["working_shifts_missing"] = (df_trip_idf_clean["working_shifts_per_week"].isna().astype(int))

#Normaliser les deux variables (0 → 1)
df_trip_idf_clean["open_days_norm"] = ((df_trip_idf_clean["open_days_per_week"] / 7)).round(2)
df_trip_idf_clean["working_shifts_norm"] = ((df_trip_idf_clean["working_shifts_per_week"] / 14)).round(2)

#Score d’accessibilité partiel
df_trip_idf_clean["accessibility_score"] = ((df_trip_idf_clean[["open_days_norm", "working_shifts_norm"]].mean(axis=1, skipna=True))).round(2)

#Fiabilité du score : 2=très fiable, 1=partiel, 0=aucune info
df_trip_idf_clean["accessibility_count"] = (df_trip_idf_clean[["open_days_norm", "working_shifts_norm"]].notna().sum(axis=1))

#Flag “accessibilité incomplète”
df_trip_idf_clean["accessibility_missing"] = (df_trip_idf_clean["accessibility_count"] < 2).astype(int)

#Mise à l’échelle finale
df_trip_idf_clean["accessibility_score_5"] = (df_trip_idf_clean["accessibility_score"] * 5).round(2)

#vérifier
df_trip_idf_clean[["open_days_per_week", "working_shifts_per_week", "open_days_norm", "working_shifts_norm",
                 "accessibility_score", "accessibility_count", "accessibility_score_5"]].head(5)

open_days_per_week  working_shifts_per_week  open_days_norm  \
74                 7.0                     13.0            1.00   
75                 5.0                      9.0            0.71   
76                 7.0                      7.0            1.00   
77                 7.0                     12.0            1.00   
78                 NaN                      NaN             NaN   

    working_shifts_norm  accessibility_score  accessibility_count  \
74                 0.93                 0.97                    2   
75                 0.64                 0.68                    2   
76                 0.50                 0.75                    2   
77                 0.86                 0.93                    2   
78                  NaN                  NaN                    0   

    accessibility_score_5  
74                   4.85  
75                   3.40  
76                   3.75  
77                   4.65  
78                    NaN

#### 4.3.e Extraire l'option Halal & kosher de la colonne special_diets (79.30% missing) --> booléenne

In [27]:
# Création de la colonne booléenne halal
df_trip_idf_clean['is_halal'] = df_trip_idf_clean['special_diets'].str.contains('Halal', case=False, na=False)
# Vérification
display(df_trip_idf_clean['is_halal'].value_counts())

# Création de la colonne booléenne kosher
df_trip_idf_clean['is_kosher'] = df_trip_idf_clean['special_diets'].str.contains('kosher', case=False, na=False)
# Vérification
display(df_trip_idf_clean['is_kosher'].value_counts())

is_halal
False    30831
True       267
Name: count, dtype: int64

is_kosher
False    31030
True        68
Name: count, dtype: int64

#### 4.3.f Transformer en booléan : "vegetarian_friendly", "vegan_options", "gluten_free"

In [33]:
#Transformation directe Y → True, N → False
cols = ["vegetarian_friendly", "vegan_options", "gluten_free"]
df_trip_idf_clean[cols] = (df_trip_idf_clean[cols].replace({'Y': True, 'N': False}).astype(bool))
df_trip_idf_clean[cols].dtypes

vegetarian_friendly    bool
vegan_options          bool
gluten_free            bool
dtype: object

# 5. DataFrame Final

In [34]:
cols_to_drop = ["atmosphere", "open_hours_per_week", "keywords", "province", "popularity_detailed", 
                "popularity_generic", "original_location", "reviews_count_in_default_language",
                "poor", "terrible", "features", "default_language", "avg_rating_missing", "total_reviews_count_missing",
                "partial_quality_missing", "open_days_missing", "working_shifts_missing", "accessibility_missing"]

df_trip_idf_clean_final = df_trip_idf_clean.drop(columns = cols_to_drop, errors="ignore")

df_trip_idf_clean_final.info()

df_trip_idf_clean_final.index = df_trip_idf_clean_final.index + 1

display(df_trip_idf_clean_final.head(3))

#Exporter en CSV
#output_path = "df_trip_idf_clean_final.csv"
#df_trip_idf_clean_final.to_csv(output_path, index = False, encoding="utf-8-sig")

<class 'pandas.core.frame.DataFrame'>
Index: 31098 entries, 74 to 155287
Data columns (total 45 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   restaurant_link                 31098 non-null  object 
 1   restaurant_name                 31098 non-null  object 
 2   country                         31098 non-null  object 
 3   region                          31098 non-null  object 
 4   city                            31094 non-null  object 
 5   address                         31098 non-null  object 
 6   latitude                        31098 non-null  float64
 7   longitude                       31098 non-null  float64
 8   claimed                         31098 non-null  object 
 9   awards                          7214 non-null   object 
 10  top_tags                        27146 non-null  object 
 11  price_level                     31098 non-null  object 
 12  price_range                     102

restaurant_link      restaurant_name country         region  \
75   g1006520-d1007522            L'imprevu  France  Ile-de-France   
76  g1006520-d10128445  La Maison du Poulet  France  Ile-de-France   
77   g1006520-d1017687         Little Tokyo  France  Ile-de-France   

                        city  \
75  Conflans Sainte Honorine   
76  Conflans Sainte Honorine   
77  Conflans Sainte Honorine   

                                              address   latitude  longitude  \
75  9 quai de la Republique, 78700 Conflans Sainte...  48.991110   2.088753   
76  91 avenue Carnot, 78700 Conflans Sainte Honori...  49.000570   2.093569   
77  2 boulevard Troussel, 78700 Conflans Sainte Ho...  48.990112   2.082339   

      claimed                                             awards  \
75    Claimed  Travellers' Choice, Certificate of Excellence ...   
76    Claimed  Travellers' Choice, Certificate of Excellence ...   
77  Unclaimed  Certificate of Excellence 2016, Certificate of...   

                                  top_tags price_level price_range  \
75                     Fine Dining, French        €€€€        None   
76  Mid-range, French, Vegetarian Friendly      €€-€€€        None   
77       Mid-range, Japanese, Sushi, Asian      €€-€€€        None   

                         meals                cuisines        special_diets  \
75  Lunch, Dinner, After-hours                  French                 None   
76               Lunch, Dinner                  French  Vegetarian Friendly   
77               Lunch, Dinner  Japanese, Sushi, Asian  Vegetarian Friendly   

    vegetarian_friendly  vegan_options  gluten_free  \
75                False          False        False   
76                 True          False        False   
77                 True          False        False   

                                  original_open_hours  open_days_per_week  \
75  {"Mon": ["10:00-15:00", "18:00-23:00"], "Tue":...                 7.0   
76  {"Mon": [], "Tue": [], "Wed": ["12:00-14:00", ...                 5.0   
77  {"Mon": ["11:30-02:30"], "Tue": ["11:30-02:30"...                 7.0   

    working_shifts_per_week  avg_rating  total_reviews_count  excellent  \
75                     13.0         4.0                292.0        1.0   
76                      9.0         4.5                296.0      202.0   
77                      7.0         4.0                204.0        3.0   

    very_good  average  food  service  value  avg_rating_filled  \
75        1.0      0.0   4.0      4.0    3.5                4.0   
76       73.0     18.0   4.5      4.5    4.5                4.5   
77        2.0      1.0   4.0      4.0    3.5                4.0   

    awards_binary  excellent_pct  very_good_pct  average_pct  \
75              1           0.34           0.34         0.00   
76              1          68.24          24.66         6.08   
77              1           1.47           0.98         0.49   

    partial_quality_score  partial_quality_count  \
75                   3.83                      3   
76                   4.50                      3   
77                   3.83                      3   

    partial_quality_score_weighted  open_days_norm  working_shifts_norm  \
75                            3.83            1.00                 0.93   
76                            4.50            0.71                 0.64   
77                            3.83            1.00                 0.50   

    accessibility_score  accessibility_count  accessibility_score_5  is_halal  \
75                 0.97                    2                   4.85     False   
76                 0.68                    2                   3.40     False   
77                 0.75                    2                   3.75     False   

    is_kosher  
75      False  
76      False  
77      False

# 6. Carte de points avec clustering

In [39]:
#Créer la carte
#m = folium.Map(location=[48.85, 2.35], zoom_start=9, tiles="OpenStreetMap")

#Ajouter le clustering
#marker_cluster = MarkerCluster().add_to(m)

#Ajouter les POI
#for _, row in df_trip_idf_clean_final.iterrows():
#    folium.Marker(location=[row["latitude"], row["longitude"]], popup=f"""<b>{row['restaurant_name']}</b><br><br><i>{row['top_tags']}<i><br><br>{row['address']}""").add_to(marker_cluster)

#m.save("map_tripadvisor_idf.html")
#m